OrbitService/SshWatchdog.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "SshWatchdog.h"
+
+#include <fcntl.h>
+#include <unistd.h>
+
+#include "OrbitBase/Logging.h"
+
+SshWatchdog::SshWatchdog(std::atomic<bool>* exit_requested)
+    : exit_requested_(exit_requested) {
+  // make stdin non blocking
+  fcntl(STDIN_FILENO, F_SETFL, O_NONBLOCK);
+};
+
+void SshWatchdog::LoopTick() {
+  const std::optional<std::string> stdin_data = ReadStdIn();
+
+  if (stdin_data) {
+    if (Active() || stdin_data.value() == start_passphrase_) {
```

Comment:
I guess we shouldn't expect that a single call from fgetc we give us exactly the start command. It could be chunked or prefixed with something else.

Reply:
I used absl::StrContains